tests/test_caiman_segmentation_extractor.py

```diff
@@ -86,6 +87,357 @@ def test_caiman_segmentation_extractor():
     for image in available_images.values():
         assert image.shape == frame_shape
 
+    # Test quality metrics (backward compatibility)
+    snr_values = extractor._get_snr_values()
+    expected_snr = np.array(
+        [
+            2.91673987,
+            1.05541609,
+            1.59315755,
+            4.66348506,
+            2.97824475,
+            2.98275578,
+            2.9776668,
+            2.94389109,
+            1.19369399,
+            1.76922279,
+        ]
+    )
+    assert_array_almost_equal(snr_values, expected_snr, decimal=6)
+    r_values = extractor._get_spatial_correlation_values()
+    expected_r_values = np.array(
+        [
+            0.35878084,
+            0.182326,
+            0.34179541,
+            0.45524581,
+            0.13731668,
+            0.58202635,
+            0.26673053,
+            0.29431159,
+            0.3361319,
+            0.34527933,
+        ]
+    )
+    assert_array_almost_equal(r_values, expected_r_values, decimal=6)
+    cnn_preds = extractor._get_cnn_predictions()
+    assert cnn_preds is None
+
+    quality_metrics = extractor.get_quality_metrics()
+    expected_metrics = ["snr", "r_values"]
+    for metric_name in expected_metrics:
+        assert metric_name in quality_metrics
+        assert len(quality_metrics[metric_name]) == 10
+    assert "cnn_preds" not in quality_metrics
+
+    # Test quality metrics as properties (new functionality)
+    property_keys = extractor.get_property_keys()
+    assert "snr" == property_keys[property_keys.index("snr")]
+    assert "r_values" == property_keys[property_keys.index("r_values")]
+    assert "cnn_preds" not in property_keys  # Should not be present since it's None
+
+    # Test getting quality metrics via property interface
+    snr_property = extractor.get_property(key="snr", ids=roi_ids)
+    assert_array_almost_equal(snr_property, expected_snr, decimal=6)
+
+    r_values_property = extractor.get_property(key="r_values", ids=roi_ids)
+    assert_array_almost_equal(r_values_property, expected_r_values, decimal=6)
+
+    # Test getting subset of quality metrics
```

Comment:
No need to test for getting a subset. This is tested in general for properties I think.